### PR TITLE
Further edit UG and DG

### PR DESCRIPTION
**Pull request by cheahTJ**
Edited more of UG in correspondence to CS2101 guidelines as well.

### CODE REVIEW

**Review by hjuntan — COMMENTED** [review]
Great changes to UG and DG!

File: docs/UserGuide.md [hunk 1]
```diff
@@ -117,48 +145,56 @@ Format: `add id/NUSID n/NAME p/PHONE_NUMBER e/EMAIL t/TAG [g/GROUP]…​`
...
 
 Format: `edit id/NUSID [n/NAME] [p/PHONE_NUMBER] [e/EMAIL] [t/TAG] [g/GROUP]` 
 
 > Note:
 > * The `NUSID` refers to the NUSID shown in the displayed person list.
 > * The `NUSID` **must be a 7-digit number following an 'E'**.
-> * > * `NAME` can consist of numbers.
+> * `NAME` can consist of numbers.
 > * `PHONE_NUMBER` has to be 3-10 digits long.
+> * `EMAIL` has to have a valid domain name (e.g. hotmail/gmail/yahoo etc).
```
[the remark this answers is not in the document]
Comment (RingoftheKing) [reply docs/UserGuide.md:178]: The email regex is as follows:
1. The local-part should only contain alphanumeric characters and these special characters (+ _  .  -)
            + The local-part may not start or end with any special characters
            + "This is followed by a '@' and then a domain name. The domain name is made up of domain labels separated by periods."
            + The domain name must:
            +     - end with a domain label at least 2 characters long
            +     - have each domain label start and end with alphanumeric characters
            +     - have each domain label consist of alphanumeric characters, separated only by hyphens, if any.

File: docs/UserGuide.md [hunk 2]
```diff
@@ -345,6 +405,7 @@ Team size: 5
 
 1. **Validity of date entered:** The current behaviour of the `schedule` command does not check for the validity of the date entered. We plan to implement a check to ensure that the date entered is valid.
 2. **Future date check:** The current behaviour of the `schedule` command does not check if the valid date entered is in the future. We plan to implement a check to ensure that the date entered is in the future.
+3. **Validity of email domain entered:** The current behaviour of the `edit` command does not check for the validity of the domain name of the email entered. We plan to implement a check to ensure that a valid domain name is entered.
```
Comment (RingoftheKing) [inline docs/UserGuide.md:408]: It actually does check for validity.

image

File: docs/UserGuide.md [hunk 3]
```diff
@@ -302,17 +361,17 @@ Format: `exit`
 
 ### Saving the data
 
-AddressBook data are saved in the hard disk automatically after any command that changes the data. There is no need to save manually.
+AronaPro data are saved in the hard disk automatically after any command that changes the data. There is no need to save manually.
 
 ### Editing the data file
 
-AddressBook data are saved automatically as a JSON file `[JAR file location]/data/addressbook.json`. Advanced users are welcome to update data directly by editing that data file.
+AronaPro data are saved automatically as a JSON file `[JAR file location]/data/addressbook.json`. Advanced users are welcome to update data directly by editing that data file.
```
[the remark this answers is not in the document]
Comment (cheahTJ) [reply docs/UserGuide.md]: Yup noted on this, I am not changing this part

File: docs/UserGuide.md (hunk 2 above)
Comment (cheahTJ) [reply docs/UserGuide.md:408]: Oh what I actually meant is that we have yet to implement a checker to check for valid domain name for `email`. As shown here, editing the email to anything random (invalid) after the `@` is accepted. The one you mentioned above is only checking the validity of the local-part so yup I think is better to include the checking of domain name as future enhancement 

Screenshot 2024-04-13 at 11 08 35 AM

Comment (cheahTJ) [reply docs/UserGuide.md:408]: Maybe I will specify it in the UG again to define what is the validity of the domain name of `email`.

File: docs/UserGuide.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (cheahTJ) [reply docs/UserGuide.md:178]: Thank you this  is really  helpful,  I will add in  to the email portion for the `EditCommand`

**Pull request merged**